int64[] p1
int64[] p2
---
int64 avr_dis